docker/build.sh

```diff
@@ -10,6 +10,7 @@ RUN_TESTS=${RUN_TESTS:-1}
 mkdir -p "$1"
 cd "$1"
 cmake -DCMAKE_BUILD_TYPE="$2" -DSTATIC_LINKING:BOOL=$STATIC_LINKING -DWARNINGS_AS_ERRORS:BOOL=$WARNINGS_AS_ERRORS ../
+find . -type f -name link.txt | xargs sed -e 's|-Wl,-Bdynamic /usr/lib/llvm8/lib/libLLVM-8.so ||' -i
```

Comment:
this is the wrong way to do this, it can be done by telling the linker not to request LLVM in Cmake, rather than having to post-process it like this, which is very brittle.

Reply:
Yes, of course. Please see the PR description for an explanation of why this hack is here. If you have a specific suggestion to fix it, I'd love to give it a try. I invested a bit of time but didn't come up with anything.

Friendly feedback: I included several caveats when I opened the PR and tried to strike a note of caution so that we don't rush into this. I share your concerns and even linked to your draft PR since I think it's the better long-term approach. Rather than just discard the work that went into this branch and the ones that came before it, though, it seemed better to share it out and decide as a group.

Reply:
> Rather than just discard the work that went into this branch and the ones that came before it, though, it seemed better to share it out and decide as a group.

Sorry if it came off that way

RE the linker issue, I believe that one of the libraries is setting LLVM as an INTERFACE_LINK_LIBRARIES property, but I didn't figure out exactly which library was doing this.

The way around this I think is to read this property and remove LLVM if its listed, then reset it like:

```
set_target_properties( LIB PROPERTIES INTERFACE_LINK_LIBRARIES ....
```

Reply:
@acj what is the content of link.txt?

I am surprised that it would still be linking to LLVM 8 if it is specifying `static` in the linker flags, it should be refusing any dynamic library : /

Unless it is *not* getting the symbols it needs from the list of libraries to statically link to. I think I ran into this problem, but I didn't have `-static` set. It could be the linker opts to not fail rather than produce a static executable?

Reply:
I made progress on this and should have something to share later today. Need to do a bit of cleanup. The gist is that both the clang and LLVM libraries are contributing to the problem via `INTERFACE_LINK_LIBRARIES`. That's likely not the root cause here, but it should give us a foothold for dealing with it early in the build process.